### ISSUE TITLE
[@intlify/vue-i18n/no-raw-text] How can I disable "[" and "]" characters?

### ISSUE THREAD

**Issue by szulcus** [intlify/eslint-plugin-vue-i18n#375]
**Tell us about your environment**

- **ESLint version:** ^8.24.0
- **eslint-plugin-vue version:** ^9.5.1
- **eslint-plugin-vue-i18n version:** ^2.0.0
- **Node version:** v16.19.0

**Please show your full configuration:**

```json5
{
	'@intlify/vue-i18n/no-raw-text': ['warn', {
		ignorePattern: '^[-#:()&%x,+]+$',
	}],
}
```

**What did you do?**

```vue
<span>[{{ <code> }}]</span>
```
![image](https://github.com/intlify/eslint-plugin-vue-i18n/assets/48328739/774d4273-620f-4c95-b419-a59cc5e63428)

**What did you expect to happen?**
Disable this rule for "[" and "]" characters.

**What actually happened?**
When I set `ignorePatters` to `^[-#:()&%x,+[]]+$` or `^[-#:()&%x,+\[\]]+$`, rule is not working.

**Repository to reproduce this issue**
I suspect it's a bug on my part. I've searched a lot on the internet, but I couldn't find a solution to this anywhere. If this is an ESLint bug and a reproduction is needed, please let me know.

**Comment by szulcus** [comment]
I used `ignoreText` as a workaround, but I'd like to include it in `ignorePattern` in case I need to add some character there in the future:
![image](https://github.com/intlify/eslint-plugin-vue-i18n/assets/48328739/116eb912-a8f7-479a-80ab-d8f238775dc0)

**Comment by ota-meshi** [comment]
If you want to escape the regular expression, you also need to escape the string.  So you need two backslashes
